### PR TITLE
[New Model] Bill of process

### LINKED ISSUE #511: [New Model]: Bill of Process (BoP)
*Resiliency Domain / Modular Production*

## Model Description

A Bill of Process (BoP) describes the best-practice approach for the production of a product. Detailed information about the processes, process precedence, and process parameters are required to manufacture the product. The exchange of process information between actors requires a standardized representation of the processes, as currently there are often separate, actor-specific forms of process representation.

This standardization effort aims to ease the understanding of process representations from other partners along the value chain by providing a shared model.
The scope of this model is a BoP that exists before actual production starts, therefore named "BoP as planned".
This model is meant to be flexible enough to accommodate a wide range of use cases and is relevant for both: information exchange (external) and internal workflows.

## Example Data

```json
{
  "productVersion": "1.4",
  "versionDataModel": "1.0",
  "billOfProcessIdentification": "https://www.example.com/bop/examplebop",
  "productName": "Barhocker",
  "processSteps": [
    {
      "processStepIdentifier": "Drehen",
      "capabilityReference": "urn:uuid:b5844423-92ac-4527-9b02-bbc64a58d193",
      "childProcessSteps": ["Pick and Place"],
      "inputMaterial": ["Sitzschale, Beine"],
      "outputMaterial": ["Barhocker"],
      "inputParameter": [
        {
          "name": "torque",
          "unit": "Nm",
          "semanticReference": "0173-1#02-AAZ808#001",
          "value": "8",
          "parameterized": "HasValue"
        }
      ],
      "outputParameter": [
        {
          "name": "weight",
          "unit": "kg",
          "semanticReference": "E-Class",
          "value": "2",
          "parameterized": "HasValue"
        }
      ],
      "precedenceRelation": [
        {
          "precedenceElements": [
            {
              "successor": ["Pick and Place"]
            }
          ]
        }
      ]
    },
    {
      "processStepIdentifier": "Pick and Place",
      "capabilityReference": "urn:uuid:a3843423-92ac-4567-1a02-bbc62a56d193",
      "inputMaterial": ["Barhocker"],
      "outputMaterial": ["Barhocker"],
      "inputParameter": [
        {
          "name": "Location",
          "parameterized": "HasNoValue"
        }
      ],
      "outputParameter": [
        {
          "name": "Location",
          "parameterized": "HasNoValue"
        }
      ]
    }
  ]
}
```

## MS1 Criteria

- [x] The proposed aspect model does not exist already
- [x] The proposed aspect model does not extend an existing aspect but introduces completely new functionality
- [x] The proposal references a Catena-X use case
- [x] Relevant standards are mentioned/linked
- [x] A first draft of the data structure is provided in an example
- [x] A modeling steward has been assigned

### PR DESCRIPTION
**Pull request by FlorianDue**
## Description

New Bill of Process Aspect Model. This model features also a README.md file to provide additional information about the models application, since it is not part of an individual standard.
 -->

Closes #511

## MS2 Criteria
(to be filled out by PR reviewer)
- [x] the model **validates** with the SAMM SDS SDK in the version specified in the Readme.md of this repository by the time of the MS2 check  (e.g., 'java -jar samm-cli.jar aspect \<path-to-aspect-model\> validate ). The  SAMM CLI is available [here](https://eclipse-esmf.github.io/esmf-developer-guide/tooling-guide/samm-cli.html) and in [GitHub](https://github.com/eclipse-esmf/esmf-sdk/releases/tag/v2.3.2)
- [x] use **Camel-Case** (e.g., "MyModelElement" or "TimeDifferenceGmtId", when in doubt follow https://google.github.io/styleguide/javaguide.html#s5.3-camel-case)
- [x] the identifiers for all model elements **start with a capital letter** except for properties
- [x] the identifier for **properties starts with a small letter**
- [x] all model elements **at least contain the fields "preferred name" and "description"** in English language. The description must be comprehensible. It is not required to write full sentences but style should be consistent over the whole model
- [x] Property and the referenced Characteristic should not have the same name
- [x] the versioning in the URN **follows semantic versioning**, where minor version bumps are backwards compatible and major version bumps are not backwards compatible. 
- [x] use **abbreviations only when necessary** and if these are sufficiently common
- [x] **avoid redundant prefixes in property names** (consider adding properties to an enclosing Entity or even adapt the namespace of the model elements, e.g., instead of having two properties `DismantlerId` and `DismantlerName` use an Entity `Dismantler` with the properties `name` and `id` or use a URN like `io.catenax.dismantler:0.0.1`)
- [x] fields `preferredName` and `description` are not the same
- [x] **`preferredName` should be human readable** and follow normal orthography (e.g., no camel case but normal word separation)
- [x] name of aspect is singular except if it only has one property which is a Collection, List or Set. In theses cases, the aspect name is plural.
- [x] units are referenced from the SAMM unit catalog whenever possible
- [x] **use constraints** to make known constraints from the use case explicit in the aspect model 
- [x] when relying on **external standards**, they are referenced through a **"see"** element
- [x] all properties with an [simple type](https://eclipse-esmf.github.io/samm-specification/2.0.0/datatypes.html) have an example value
- [x] metadata.json exists with status "release"
- [x] generated json schema validates against example json payload
- [x] file RELEASE_NOTES.md exists and contains entries for proposed model changes 
- [x] all contributors to this model are mentioned in copyright header of model file

## MS3 Criteria
(to be filled out by semantic modeling team before merge to main-branch)
- [x] All required reviewers have approved this PR (see reviewers section)
- [x] The new aspect (version) will be implemented by at least one data provider
- [x] The new aspect (version) will be consumed by at least one data consumer
- [x] There exists valid test data
- [x] In case of a new (incompatible) major version to an existing version, a migration strategy has been developed
- [x] The model has at least version '1.0.0'
- [x] If a previous model exists, model deprecation has been checked for previous model
- [x] The release date in the Release Note is set to the date of the MS3 approval

### CODE REVIEW

File: io.catenax.shared.bill_of_process/1.0.0/BillOfProcessSharedAspect.ttl [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+#
+# This work is made available under the terms of the
+# Creative Commons Attribution 4.0 International (CC-BY-4.0) license,
+# which is available at
+# https://creativecommons.org/licenses/by/4.0/legalcode.
+#
+# SPDX-License-Identifier: CC-BY-4.0
+##########################################################################################
+@prefix samm: <urn:samm:org.eclipse.esmf.samm:meta-model:2.1.0#>.
+@prefix samm-c: <urn:samm:org.eclipse.esmf.samm:characteristic:2.0.0#>.
```
Comment (agg3fe) [inline io.catenax.shared.bill_of_process/1.0.0/BillOfProcessSharedAspect.ttl:17]: please use 2.1.0 samm version

Comment (FlorianDue) [reply io.catenax.shared.bill_of_process/1.0.0/BillOfProcessSharedAspect.ttl]: updated

**Comment by FlorianDue** [comment]
The model also requires the Manufacturing Capability Model Version 3.0.0. It has an open PR #574

**Review by agg3fe — APPROVED** [review]
LGTM

**Comment by agg3fe** [comment]
please check eclipsefdn/eca step.

**Comment by FlorianDue** [comment]
> please check eclipsefdn/eca step.

what do you mean by that?

**Pull request merged**